### PR TITLE
Create MagicNumber.c

### PR DESCRIPTION
**Pull request by GauriKashyap**

### CODE REVIEW

**Review by hitaarthh — COMMENTED** [review]
I guess the code has some runtime error, kindly look into .

**Review by hitaarthh — COMMENTED** [review]
remove extra spaces

**Review by GauriKashyap — COMMENTED** [review]
removed extra spaces

File: MagicNumber.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    
+    // assigning the number to temporary variable
+    temp = n;   
+      
+    // calculating the sum of digits  
+    while ( temp > 0)  
+    {  
+        // extracting digits and calculating the sum  
+        sum = sum + temp % 10;   
+        temp = temp / 10;
```
Comment (hitaarthh) [inline MagicNumber.c:16]:
```suggestion
        temp/= 10;  
```

File: MagicNumber.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    temp = sum;   
+    printf (" \n digit sum = %d", temp);  
+      
+    // reverse sum of given digits  
+    while ( temp > 0)  
+    {  
+        rev = rev * 10 + temp % 10;  
+        temp = temp / 10;  
+    }  
+
```
Comment (hitaarthh) [inline MagicNumber.c]:
```suggestion
```

**Review by hitaarthh — CHANGES_REQUESTED** [review]
Check your code again

**Review by hitaarthh — APPROVED** [review]

**Pull request merged**